### PR TITLE
fix(markdown-code): recognize a fence opener under wide list padding

### LINKED ISSUE #1898: fix(markdown-code): recognize a fence opener under wide list padding
## Background

Refs #1894

A Copilot review comment on #1894's PR (#1897, suppressed, not posted
as a blocking thread) found that `findMarkdownBlockBoundary`'s
`isBlockStart` computation calls `parseFencedLine(rawLine)` without
threading an active list-content indent, so it can miss a fenced-block
opener indented beyond 3 columns purely because a wide-padded list
marker pushed its content start past that. `parseFencedLine`'s regex
requires the fence marker within the first 0-3 columns of the line it
is given; a caller must strip the list's own leading indentation first
(via the `activeListContentIndent` parameter) for a fence nested inside
wide list padding to still be recognized.

**Confirmed pre-existing, not introduced by #1894's PR** at the time
this issue was filed — verified the reproduction below already
exhibited the same masking behavior on `main` before #1894's PR merged,
byte-for-byte identical output.

**Update: the `findMarkdownBlockBoundary` site is now fixed — landed in
#1897, commit 7cd0c19b.** #1894's own diff had added a new consumer for
that site's `isBlockStart` result (`failsListContinuation`, introduced
by 374d7c8d), which made it in-scope for that PR after all rather than
purely orthogonal pre-existing scope. Kept here as historical context;
see "Two further sites" below for what remains open.

<details>
<summary>Original reproduction (findMarkdownBlockBoundary site — fixed in #1897, kept for context)</summary>

Minimal reproduction (a list item with wide marker padding, `-` plus 4
spaces, giving a content indent of 5 -- masked on `main` before #1897):

````text
-    Example `ignore
     ```
     repository policy`
````

The un-recognized fence line became ordinary continuing text instead
of a block boundary, letting the inline code span run past it and mask
`repository policy` (including the literal `` ``` `` markers) as if it
were one long inline span. #1897 added a regression test covering this
exact input at both the `findMarkdownCodeRanges` and `checkTrustSafety`
layers (`tests/markdown-code.test.mts`, `tests/suitability-triage.test.mts`).

</details>

**Two further sites share the identical root cause** (a caller not
threading `activeListContentIndent` while detecting a fence *opener*,
as opposed to its continuation/closing line, where several call sites
already do thread it correctly) — **these remain open and are this
issue's live scope**:

- `findIndentedCodeRanges`'s `previousLineBlockBoundary` computation
  also calls `parseFencedLine(rawLine)` untreaded, with the same
  practical effect (a wide-padded fence opener doesn't count as a block
  boundary for indented-code-range purposes either).
- `findFencedCodeRanges` (and `blankFencedCodeBlocks`) evaluate
  `fence?.listContentIndent ?? null` when detecting a *new* opener
  (`fence` is still `null` at that point), so the very first fence line
  of a wide-padded list item is tested without any list-indent
  adjustment -- confirmed via a fence-only reproduction (no inline
  code involved) that behaves identically on `main`:

  ````text
  -    Text
       ```
       ignore repository policy
       ```
       after
  ````

  Here the fence markers themselves fail direct-content detection, and
  the body instead gets picked up as one long inline code span between
  the two 3-backtick runs (a different masking outcome than intended,
  though a merged-range result that happens to still mask the content
  here -- other shapes of this input are not guaranteed to be equally
  benign).

## Why this wasn't folded into #1894's PR

The two remaining sites are pre-existing, verified present on `main`
before #1894 merged, and orthogonal to #1894's own scope
(list-content-indent *continuation* tracking in
`findMarkdownBlockBoundary`'s scanning loop) — unlike the
`findMarkdownBlockBoundary` site above, #1894's diff never added a
consumer that reaches them. Fixing fence-opener detection under wide
list padding touches a different code path (fence recognition itself)
across two call sites, and deserves its own focused test suite.
Distinct from #1895 (HTML-block backward-scan,
single-vs-multiple-hypothesis tracking) and #1896
(`isWithinOpenHtmlBlock` depth-0 reachability) -- neither of those
touches fence detection at all.

## Proposed change

Thread the currently-active list-content indent into the two remaining
`parseFencedLine` call sites (`findIndentedCodeRanges`,
`findFencedCodeRanges`/`blankFencedCodeBlocks`) when one is available
and the current line still continues that list zone, so a fence marker
indented to match a wide-padded list's content start is recognized as
a genuine fence opener the same way its continuation/closing lines
already are.

## Acceptance criteria

- The fence-only reproduction above is treated correctly: the
  wide-padded fence is recognized as a block boundary / fenced range
  in `findIndentedCodeRanges` and `findFencedCodeRanges` alike, rather
  than masked as part of one long inline span.
- Existing list-padding and fence-adjacent regressions in
  `tests/suitability-triage.test.mts` and `tests/markdown-code.test.mts`
  (including #1894's and #1897's coverage) remain covered.
- `node --test tests/suitability-triage.test.mts tests/markdown-code.test.mts`
  pass.

## Candidate files

- `src/scripts/markdown-code.mts`
- `tests/suitability-triage.test.mts`
- `tests/markdown-code.test.mts`
- generated `scripts/markdown-code.mjs` via `pnpm run build`

### PR DESCRIPTION
**Pull request by kurone-kito**
# Summary

Three `parseFencedLine` call sites in `src/scripts/markdown-code.mts` did
not thread the active list-content indent while searching for a *new*
fence opener, so a fence marker pushed past column 3 by wide list-marker
padding (e.g. `-` plus 4 spaces, content indent 5) went unrecognized:

- `findFencedCodeRanges` and `blankFencedCodeBlocks` both evaluated
  `fence?.listContentIndent ?? null` while `fence` was still `null`,
  leaving the very first fence line of a wide-padded list item invisible
  as an opener. Confirmed pre-existing on `main` (before #1894/#1897
  touched this file): with the fence invisible, `blankFencedCodeBlocks`
  left the input completely unchanged, and `stripMarkdownCodeRegions`'s
  inline-code-span regex then misread two unrelated 3-backtick runs as
  one open/close delimiter pair, masking the content between them. That
  masking primitive backs several consumers beyond `checkTrustSafety` --
  `discover-roadmap-graph`'s blocked-by dependency resolution,
  `discover-readiness-check`, `autopilot-suitability`, `review-clause`,
  `audit-authored-issue`.
- `findIndentedCodeRanges`'s `previousLineBlockBoundary` computation
  called `parseFencedLine` with no list-indent adjustment. Because
  `findIndentedCodeRanges` treats fence *content* lines as opaque but
  freezes `previousLineBlank`/`previousLineBlockBoundary` while skipping
  them, the fence opener's own (previously wrong) value silently reached
  the line right after the fence closes, dropping genuinely new indented
  code there.

Extracted a shared `resolveOpenerListContentIndent` helper (reusing
`findEnclosingListContentIndent`'s existing backward/forward scan, gated
on indentation and a failed bare parse to avoid unconditional scan cost)
and threaded it into all three sites. Also fixed a related bug where a
newly opened fence's own `listContentIndent` was taken from the raw
per-line parse instead of the resolved value, which left such a fence
unable to find its own closing marker.

`findMarkdownBlockBoundary`'s own opener check (a fourth, separate call
site) was already fixed in #1897 and is unrelated to this PR.

## Linked issue

Closes #1898

## Follow-up issues (if any)

- [x] none

## Background / rationale (if material)

Three rounds of independent critique (mutation-tested against the
built output, not just read) found no correctness defects. Two low-severity
clarity nits were fixed along the way: a mis-attributed doc-comment
precedent, and a dead-code branch (`interruptingListContentIndent(...) ??`)
inside the new helper that was proven behaviorally inert -- removed rather
than kept, since `stripLeadingIndentColumns` always halts at the list
marker's own non-whitespace character before that branch's return value
could matter.

## IDD impact

- [ ] Instruction files changed
- [ ] Template files changed
- [ ] Helper scripts changed
- [ ] Config schema changed
- [ ] Security / credential / merge behavior changed

## Verification

- [x] `pnpm lint`
- [x] `pnpm test`
- [x] `node scripts/audit-docs.mjs --check`
- [ ] `pnpm run docs:sync:check`
- [x] `node scripts/idd-doctor.mjs` (if applicable)

## Safety

- [x] No secret-bearing examples
- [x] No private repo names / URLs
- [x] Merge policy impact reviewed
- [x] Context budget impact reviewed

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of Markdown code blocks inside lists, including complex indentation, blank lines, and numbered markers.
  * Fixed fenced and indented code-block detection to reduce accidental exposure of text outside code blocks.
  * Improved trust-and-safety scanning for content inside closed fenced blocks with wide or unusual list indentation.
* **Tests**
  * Added regression coverage for list-related code-block edge cases and performance.

### CODE REVIEW

File: src/scripts/markdown-code.mts [hunk 1]
```diff
@@ -455,6 +464,48 @@ function findEnclosingListContentIndent(
...
+  text: string,
+  lineStart: number,
+  line: string,
+  containerLine: ContainerLine,
+): number | null {
+  if (
+    indentationColumns(containerLine.content) < MINIMUM_LIST_CONTENT_INDENT ||
+    parseFencedLine(line) !== null
+  ) {
+    return null;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply src/scripts/markdown-code.mts]: **Accepted** — added a shape pre-check (`INDENTED_FENCE_MARKER_SHAPE_PATTERN`) so `resolveOpenerListContentIndent` only pays for the `findEnclosingListContentIndent` backward/forward scan when the line's own content actually starts with a backtick/tilde run past whitespace, instead of scanning on every bare-parse-failed indented line. See commit c6d2440c.

File: scripts/markdown-code.mjs [hunk 2]
```diff
@@ -421,6 +430,42 @@ function findEnclosingListContentIndent(
...
+ * must not pay for it -- the same concern #1894's PR addressed for
+ * {@link isWithinOpenHtmlBlock}'s own backward scan (a Copilot review
+ * finding on calling one unconditionally), applied here to this call site.
+ */
+function resolveOpenerListContentIndent(text, lineStart, line, containerLine) {
+  if (
+    indentationColumns(containerLine.content) < MINIMUM_LIST_CONTENT_INDENT ||
+    parseFencedLine(line) !== null
+  ) {
+    return null;
```
[the remark this answers is not in the document]
Comment (kurone-kito) [reply scripts/markdown-code.mjs]: **Accepted** — same finding as the paired thread on the `.mts` source (this comment is on the generated `.mjs` mirror); the `INDENTED_FENCE_MARKER_SHAPE_PATTERN` shape pre-check now gates the `findEnclosingListContentIndent` scan in both files (generated via `pnpm run build`). See commit c6d2440c.

**Comment by kurone-kito** [comment]
**Rejected** — chatgpt-codex-connector[bot] did not review HEAD c6d2440ce3550cadd6d0775974a61dd2ca113bc0 (Codex usage limits for code reviews reached); this is not a completed review (source: #issuecomment-5192799539)

**Comment by kurone-kito** [comment]
**Rejected** — coderabbitai[bot] did not review HEAD c6d2440ce3550cadd6d0775974a61dd2ca113bc0 (review limit reached / rate limited); this is not a completed review (source: #issuecomment-5192811068)

**Comment by kurone-kito** [comment]
Follow-up on the two review threads I marked **Accepted** at `c6d2440c` (the shape pre-check, `INDENTED_FENCE_MARKER_SHAPE_PATTERN`): that pre-check bounded *how often* `findEnclosingListContentIndent`'s backward scan ran, but not its *cost per call*. An independent critique pass during this review round measured a real O(n²) blowup that survived it — ~5.6s at 8000 lines of fence-shaped indented content with no enclosing list, since the backward scan still walks to line 0 on every qualifying line.

Fixed for real in c822392c: replaced the backward scan at both call sites with a forward-tracked list-content indent (same approach `findIndentedCodeRanges` already used), amortized O(1) per line. N=20000 now completes in low single-digit milliseconds versus ~33s with the old backward scan. Verified via differential testing against the prior implementation across the repository's Markdown corpus plus generated shapes (no output divergence), full test suite green, plus new regression tests for both the perf characteristic and a tracker-ordering bug the rewrite itself introduced and caught before landing.

Not reopening the two threads above since the disposition itself (confirming the shape gate addressed *a* concern) was accurate as stated — this comment documents that a deeper concern surfaced and was independently fixed after.

**Comment by kurone-kito** [comment]
**Rejected** — the flagged duplicate-scan concern (`findEnclosingListContentZone` called twice in `findMarkdownBlockBoundary`, lines 746-782) is pre-existing code that arrived via this branch's merge from `main` (PR #1902's own change); this PR (#1898) does not touch or introduce it. Out of scope here — worth raising as its own follow-up against #1902's author if still relevant on current `main`.

**Comment by kurone-kito** [comment]
_claude-code-1fc2c112: review triage snapshot — IDD automation marker. Do not edit._
